src/hotspot/share/gc/shared/memAllocator.cpp

```diff
@@ -187,9 +206,8 @@ void MemAllocator::Allocation::notify_allocation_jvmti_sampler() {
     return;
   }
 
-  if (!_allocated_outside_tlab && _allocated_tlab_size == 0 && !_tlab_end_reset_for_sample) {
-    // Sample if it's a non-TLAB allocation, or a TLAB allocation that either refills the TLAB
-    // or expands it due to taking a sampler induced slow path.
+  if (!should_notify_allocation_no_jvmti_vmobjalloc()) {
+    // Called here only for JVMTI VMObjectAlloc event
```

Comment:
This comment sounds like we are executing the return statement because of `JVMTI VMObjectAlloc event`, but isn't it the opposite? The reason why we are in this function and *not* executing the return statement is because the VMObjectAlloc event is turned on. Would it make more sense to move the comment to after this if block?